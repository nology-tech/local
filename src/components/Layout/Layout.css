/*                          Colours                         */
/*                          Font stuff              */
.layout {
  height: 100vh;
  width: 100vw;
  background: rgb(236, 148, 114);
  background: linear-gradient(152deg, rgb(236, 148, 114) 0%, rgb(224, 95, 30) 100%);
}/*# sourceMappingURL=Layout.css.map */