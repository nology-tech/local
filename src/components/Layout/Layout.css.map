{"version":3,"sources":["../../styles/_variables.scss","Layout.scss","../../styles/_mixins.scss","Layout.css"],"names":[],"mappings":"AAOA,6DAAA;AAuBA,qDAAA;AC3BA;EACE,aAAA;EACA,YAAA;ECKA,8BFIW;EEHX,iFFIoB;AGRtB","file":"Layout.css"}