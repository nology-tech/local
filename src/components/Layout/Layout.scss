@use "../../styles/variables" as *;
@use "../../styles/mixins" as *;

.layout {
  height: 100vh;
  width: 100vw;
  @include set-background;
  @include text-normal;
}
