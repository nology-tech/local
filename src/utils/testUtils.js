import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
