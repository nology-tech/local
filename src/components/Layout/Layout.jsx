import "./Layout.scss";
import Menu from "../Menu/Menu";
import MenuIcon from "../MenuIcon/MenuIcon";
import { useState } from "react";

const Layout = ({ children, isWithMenu }) => {
  if (!isWithMenu) {
    return <div className="layout">{children}</div>;
  }

  //Layout with Menu Icon
  const [showMenu, setShowMenu] = useState(false);

  const toggleMenu = () => {
    setShowMenu(!showMenu);
    console.log("here");
  };

  return (
    <>
      <div className="layout">
        {showMenu && (
          <>
            <div className="menu__shadow"></div>
            <div className="layout__menu">
              <Menu toggleMenu={toggleMenu} />
            </div>
          </>
        )}
        {!showMenu && <MenuIcon toggleMenu={toggleMenu} />}
        {children}
      </div>
    </>
  );
};
export default Layout;
