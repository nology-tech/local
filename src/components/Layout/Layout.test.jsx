import Layout from "./Layout";
import { customRender } from "../../utils/testUtils";
import userEvent from "@testing-library/user-event";

it("should render the layout component", () => {
  const { container } = customRender(
    <Layout>
      <p>Hello there</p>
    </Layout>
  );

  expect(container).toMatchSnapshot();
});

it("should apply a different class when the menu is open/closed", () => {
  const { container } = customRender(<Layout isWithMenu={true} />);

  const menu = container.querySelector(".layout__menu");

  expect(menu).toHaveClass("menu-closed");

  const toggleButton = container.querySelector(".menu-icon__container");

  userEvent.click(toggleButton);

  expect(menu).toHaveClass("menu-open");
});
