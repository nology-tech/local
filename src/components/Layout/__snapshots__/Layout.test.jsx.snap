// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the layout component 1`] = `
<div>
  <div
    class="layout"
  >
    <p>
      Hello there
    </p>
  </div>
</div>
`;
